iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTagExec.java

```diff
@@ -64,4 +64,13 @@ public static void dropTag(Table table, AlterTableSnapshotRefSpec.DropSnapshotRe
       table.manageSnapshots().removeTag(tagName).commit();
     }
   }
+
+  public static void replaceTag(Table table, AlterTableSnapshotRefSpec.ReplaceSnapshotrefSpec replaceTagRefSpec) {
+    String sourceTag = replaceTagRefSpec.getSourceRefName();
+    long targetSnapshot = replaceTagRefSpec.getTargetSnapshot();
+    LOG.info("Replacing tag {} with snapshot {} on iceberg table {}", sourceTag, targetSnapshot, table.name());
+    ManageSnapshots manageSnapshots = table.manageSnapshots().replaceTag(sourceTag, targetSnapshot);
+    IcebergBranchExec.setOptionalReplaceParams(replaceTagRefSpec, manageSnapshots, sourceTag);
```

Comment:
I know this line want to reuse `IcebergBranchExec `code to reduce code duplication. But my original thought was to keep Branch and Tag operation separate.

I am not aginst doing this as Branch and Tag operation indeed have many similar code.
1) Maybe we can merge the two  `IcebergBranchExec ` & `IcebergTagExec ` into one class. 
2) Or we can extract the common codes and put it into the utility class. 
This optimization can be done once branch&tag feature are all done in the future. :)

Reply:
If I got you right, I will do [1] in the next iter & merge the classes as suggested as ``IcebergSnapshotRefExec``